TimeParse.cpp

```diff
@@ -0,0 +1,75 @@
+#include "TimeParse.h"
+
+using namespace std;
+
+TimeParse::TimeParse(string line) {
+    doParsing(line);
+}
+void TimeParse::doParsing(string input) {
+    int set_count =0, startIndex =0;
+    string word;
+    for(int i=0;i<input.length();i++)
+    {
+        if(set_count == 3) {
+            word = input.substr(startIndex,(input.length()-1)-startIndex);
+            set_count++;
+            setMemberVariableValues(word,set_count);
+            break;
+        }
+        if(input[i] ==':'){
+            word = input.substr(startIndex,i-startIndex);
+            startIndex=i+1;
```

Comment:
startIndex=i+1; to be done before assigning word

Reply:
if (input[i] == ' ') {
            word = input.substr(startIndex, i - startIndex);
            while(input[++i] == ' ');
            startIndex = i;
            set_count++;
            i--;
            setMemberVariableValues(word, set_count);
        }
        
        
        what about the startIndex in this snippet

Reply:
here also

Reply:
Sorry, Ignore this comment